packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js

```diff
@@ -106,29 +104,36 @@ export default class ExchangeDomainDkimAutoconfigCtrl extends DkimAutoConfigurat
   activateDkim() {
     const dkimSelectors = this.domain.dkim;
 
-    this.services.ExchangeDomains.enableDkim(
+    const promise = this.services.ExchangeDomains.enableDkim(
       this.$routerParams.organization,
       this.$routerParams.productId,
       this.domain.name,
       dkimSelectors[0].selectorName,
-    ).finally(() => {
-      this.services.navigation.resetAction();
-    });
+    );
+    this.handleDkimOperationResponse(promise);
   }
 
   deactivateDkim() {
     const dkimSelectors = this.domain.dkim;
 
-    this.services.ExchangeDomains.disableDkim(
+    const promise = this.services.ExchangeDomains.disableDkim(
       this.$routerParams.organization,
       this.$routerParams.productId,
       this.domain.name,
       dkimSelectors[0].status === this.DKIM_STATUS.IN_PRODUCTION
         ? dkimSelectors[0].selectorName
         : dkimSelectors[1].selectorName,
-    ).finally(() => {
-      this.services.navigation.resetAction();
-    });
+    );
+    this.handleDkimOperationResponse(promise);
+  }
+
+  resetAction(reload = false) {
+    this.services.navigation.resetAction();
+    if (reload) {
+      this.services.$state.go('exchange.dashboard.domain').then(() => {
```

Comment:
Maybe you can try to use the $state.go reload option ? We did this in Octavia and it works well :https://github.com/ovh/manager/blob/feat/octavia-batch-1/packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/create/routing.js#L19